python/caffe/_caffe.cpp

```diff
@@ -252,6 +252,7 @@ BOOST_PYTHON_MODULE(_caffe) {
     .def("_set_input_arrays", &Net_SetInputArrays,
         bp::with_custodian_and_ward<1, 2, bp::with_custodian_and_ward<1, 3> >())
     .def("save", &Net_Save);
+  bp::register_ptr_to_python<shared_ptr<Net<Dtype> > >();
```

Comment:
@longjon can you comment on this? It seems redundant to me given the `bp::class_` declarations but boost 1.60 complains without it.